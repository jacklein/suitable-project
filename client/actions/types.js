// badge actions
export const FETCH_BADGES = 'fetch_badges';
export const UPDATE_PROGRESS = 'update_progress';