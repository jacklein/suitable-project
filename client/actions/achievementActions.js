import {
  FETCH_ACHIEVEMENTS
} from './types';
import axios from 'axios';

export const fetchAchievements = () => async dispatch => {
  const res = await axios.get('http://192.168.1.137:3001/v1/achievements');
  dispatch({ type: FETCH_ACHIEVEMENTS, payload: res.data });
}