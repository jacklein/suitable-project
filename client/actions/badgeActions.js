import {
  FETCH_BADGES, 
  UPDATE_PROGRESS
} from './types';
import axios from 'axios';

const URI = 'http://192.168.1.137:3001'

/**
 * res.data structure:
 * [
 * { 'id': 123, 'details': {...}, 'relationships': {...} },
 * { 'id': 456, 'details': {...}, 'relationships': {...} } 
 * ]
 */
export const fetchBadges = () => async dispatch => {
  try{
    const res = await axios.get(`${URI}/v1/achievements`);
    dispatch({ type: FETCH_BADGES, payload: res.data });
  } catch(e) {
    console.log(e);
  }
}

/**
 * res.data structure:
 * {
 * 'id': (id of completed activity),
 * 'updates': [{'achievementID': 456, 'progress': 0.5}, {...}, {...}]
 * }
 */
export const updateProgress = (id, callback) => async dispatch => {
  try {
    const res = await axios.get(`${URI}/v1/activities/${id}/complete`);
    dispatch({ type: UPDATE_PROGRESS, payload: res.data.updates });
    callback();
  } catch(e) {
    console.log(e);
  }
}