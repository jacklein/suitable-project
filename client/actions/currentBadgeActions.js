import {
  SET_INDEX
} from './types';

/**
 * Sets the index of the badge currently being looked at
 */
export const setIndex = index => async dispatch => {
  dispatch({ type: SET_INDEX, payload: index });
}