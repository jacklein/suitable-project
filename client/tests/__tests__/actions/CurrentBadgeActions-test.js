import { setIndex } from '../../../actions/currentBadgeActions';
import { SET_INDEX } from '../../../actions/types';

describe('CurrentBadgeActions', () => {
  it('creates a SET_INDEX action', () => {
    expect(setIndex(1)).toEqual(
      {
        type: SET_INDEX,
        payload: 1
      }
    );

    expect(setIndex(1)).toMatchSnapshot();
  });
});