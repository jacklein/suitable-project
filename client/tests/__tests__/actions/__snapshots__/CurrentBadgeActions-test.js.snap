// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentBadgeActions creates a SET_INDEX action 1`] = `
Object {
  "payload": 1,
  "type": "set_index",
}
`;

exports[`creates a SET_INDEX action 1`] = `
Object {
  "payload": 1,
  "type": "set_index",
}
`;
